fava/core/ingest.py

```diff
@@ -0,0 +1,92 @@
+"""Ingest helper functions."""
+
+import os
+import runpy
+import tempfile
+from collections import defaultdict, namedtuple
+
+from beancount.core import data
+from beancount.ingest import identify, cache, extract
+
+from fava.core.helpers import FavaModule
+
+IngestError = namedtuple('IngestError', 'source message entry')
+
+
+# pylint: disable=too-few-public-methods
+class IngestModule(FavaModule):
+    """Exposes ingest functionality."""
+
+    def __init__(self, ledger):
+        super().__init__(ledger)
+        self.config = None
+        self.ingest_dirs = []
+        self.uploaded_files = defaultdict(list)
```

Comment:
This should just be list (not a dict of lists). So far all application state is either global or relative to a ledger and so should this be IMHO. Using sessions for this just seems to lead to unnecessary complication.